pymongo/change_stream.py

```diff
@@ -157,15 +157,19 @@ def _process_result(self, result, session, server, sock_info, slave_ok):
         version in order to determine whether to cache this value.
         """
         if not result['cursor']['firstBatch']:
-            if (self._start_at_operation_time is None and
-                    self.resume_token is None and
+            if ('postBatchResumeToken' not in result['cursor'] and
+                    self._start_at_operation_time is None and
+                    self._uses_resume_after is False and
+                    self._uses_start_after is False and
                     sock_info.max_wire_version >= 7):
                 self._start_at_operation_time = result.get("operationTime")
                 # PYTHON-2181: informative error on missing operationTime.
                 if self._start_at_operation_time is None:
                     raise OperationFailure(
                         "Expected field 'operationTime' missing from command "
                         "response : %r" % (result, ))
+            elif 'postBatchResumeToken' in result['cursor']:
+                self._resume_token = result['cursor']['postBatchResumeToken']
```

Comment:
I suggest moving this case (`'postBatchResumeToken' in result['cursor']`) to be the first `if` so we don't need to repeat the `postBatchResumeToken` check in both.